### ISSUE TITLE
Transfer learning with different classes

### ISSUE THREAD

**Issue by RailwaysPi** [PingoLH/FCHarDNet#12]
I would like to take a pre-trained Cityscapes model for transfer learning with a dataset that does not have the same amount of classes. At the moment, this causes an error in the module method _load_state_dict()_. What do I have to change in the code to not initialize the last network layer for making it work?

**Comment by PingoLH** [comment]
Hi, you can try to modify the model in train.py after the load_state_dict() by:
model.finalConv  = nn.Conv2d(48,  out_channels=new_classes, kernel_size=1, stride=1,
                 padding=0, bias=True)

**Comment by RailwaysPi** [comment]
Thanks, that works for me. Just consider to place the layer on the correct device. I implemented following method in the hardnet class for reinitializing the last layer:
```
def init_last_layer(self, n_classes):
    self.finalConv = nn.Conv2d(48, n_classes, kernel_size=1, stride=1, padding=0, bias=True).cuda()
```
Please let me know, if there are improvements for this code!
